nova_vm/src/ecmascript/abstract_operations/type_conversion.rs

```diff
@@ -829,15 +809,48 @@ pub(crate) fn to_big_int(
 }
 
 /// ### [7.1.14 StringToBigInt ( str )](https://tc39.es/ecma262/#sec-stringtobigint)
-pub(crate) fn string_to_big_int(_agent: &mut Agent, _argument: String) -> Option<BigInt> {
+pub(crate) fn string_to_big_int(
+    agent: &mut Agent,
+    gc: GcScope<'_, '_>,
+    argument: String,
+) -> JsResult<BigInt> {
     // 1. Let text be StringToCodePoints(str).
     // 2. Let literal be ParseText(text, StringIntegerLiteral).
     // 3. If literal is a List of errors, return undefined.
     // 4. Let mv be the MV of literal.
     // 5. Assert: mv is an integer.
+    let base_mv = argument.as_str(agent); // Extra line literally just for displaying error
```

Comment:
**suggestion:** It would be useful to reference some of the StringIntegerLiteral parsing rules for the steps below. Right now as I read this, it looks absolutely reasonable but I don't know if eg. the StringIntegerLiteral parsing rules have a special case for parsing `deadbeef` as `0xdeadbeef` hex number, or something like that.

Reply:
Not exactly sure if I've done what you were looking for but linked to the semantics outlined in the docs and tried to piece up what I have done to these.